website/static/js/projectSettings.js

```diff
@@ -8,77 +8,114 @@ var $osf = require('js/osfHelpers');
             self.MESSAGE_ERROR_CLASS = 'text-danger';
 
-            if (!updateUrl) {
+            if (!params.updateUrl) {
                 throw new Error(self.INSTANTIATION_ERROR_MESSAGE);
             }
 
-            self.categories = categories;
-            self.category = ko.observable(category);
-            self.updateUrl = updateUrl;
-
-            self.selectedCategory = ko.observable(category);
-            self.dirty = ko.observable(false);
-            self.selectedCategory.subscribe(function(value) {
-                if (value !== self.category()) {
-                    self.dirty(true);
-                }
-            });
+            self.categories = params.categories;
+            self.category = params.category;
+            self.updateUrl = params.updateUrl;
+            self.selectedCategory = ko.observable(params.category);
         },
-        updateSuccess: function(newcategory) {
+
+        /*success handler*/
+        updateSuccess: function() {
             var self = this;
             self.changeMessage(self.UPDATE_SUCCESS_MESSAGE, self.MESSAGE_SUCCESS_CLASS);
-            self.category(newcategory);
-            self.dirty(false);
         },
-        updateError: function(xhr, status, error) {
+
+        /*error handlers*/
+        updateCategoryError: function(xhr, status, error) {
             var self = this;
-            self.changeMessage(self.UPDATE_ERROR_MESSAGE, self.MESSAGE_ERROR_CLASS);
-            Raven.captureMessage(self.UPDATE_ERROR_MESSAGE_RAVEN, {
+            self.changeMessage(self.UPDATE_CATEGORY_ERROR_MESSAGE, self.MESSAGE_ERROR_CLASS);
+            Raven.captureMessage(self.UPDATE_CATEGORY_ERROR_MESSAGE_RAVEN, {
                 url: self.updateUrl,
                 textStatus: status,
                 err: error
             });
         },
+        updateDescriptionError: function() {
+            var self = this;
+            self.changeMessage(self.UPDATE_DESCRIPTION_ERROR_MESSAGE, self.MESSAGE_ERROR_CLASS);
+        },
+
+        /*update handlers*/
         updateCategory: function() {
             var self = this;
             return $osf.putJSON(self.updateUrl, {
                     category: self.selectedCategory()
                 })
                 .then(function(response) {
+                    self.category = self.selectedCategory();
                     return response.updated_fields.category;
                 })
-                .done(self.updateSuccess.bind(self))
-                .fail(self.updateError.bind(self));
+                .done(self.updateTitle.bind(self))
+                .fail(self.updateCategoryError.bind(self));
+        },
+        updateTitle: function() {
+            var self = this;
+            return $osf.putJSON(self.updateUrl, {
+                    title: self.title()
+                })
+                .done(function() {
+                    self.decodedTitle = self.title();
+                },
+                self.updateDescription.bind(self));
```

Comment:
In general, perform front-end validation as appropriate. The user shouldn't have to rely on a server response to find out that their field is a problem, and if we do expect some sort of edge case validation, rely on an informative server response to help trace the error. Don't try to validate the really insane edge cases- at some point you're not responsible for users doing totally off the wall.

(Sending three requests is rather a kludge- good design can help prevent that sort of thing from being necessary)